### ISSUE TITLE
AggressiveSplittingPlugin with recordsPath set in production mode compile error the second time

### ISSUE THREAD

**Issue by AMDvsTMD** [webpack/webpack#7525]
# Bug report
as the issuse [7520]: https://github.com/webpack/webpack/issues/7520 descibed when update to the lasted version in development mod problems solved but in production mod some new error triggered

**What is the current behavior?**
when compile the second time in production mode error triggered stack:
```
			for (const refModule of blockInfo.modules) {
			                                  ^

TypeError: Cannot read property 'modules' of undefined
    at Compilation.processDependenciesBlocksForChunkGroups (D:\pdd_web_personal\node_modules\webpack\lib\Compilation.js:1318:38)
    at Compilation.seal (D:\pdd_web_personal\node_modules\webpack\lib\Compilation.js:883:8)
    at hooks.optimizeAssets.callAsync.err (D:\pdd_web_personal\node_modules\webpack\lib\Compilation.js:984:20)
    at AsyncSeriesHook.eval [as callAsync] (eval at create (D:\pdd_web_personal\node_modules\tapable\lib\HookCodeFactory.js:24:12), <anonymous>:6:1)
    at hooks.optimizeChunkAssets.callAsync.err (D:\pdd_web_personal\node_modules\webpack\lib\Compilation.js:977:32)
    at _err0 (eval at create (D:\pdd_web_personal\node_modules\tapable\lib\HookCodeFactory.js:24:12), <anonymous>:11:1)
    at D:\pdd_web_personal\node_modules\uglifyjs-webpack-plugin\dist\index.js:334:11
    at step (D:\pdd_web_personal\node_modules\uglifyjs-webpack-plugin\dist\uglify\index.js:90:11)
    at done (D:\pdd_web_personal\node_modules\uglifyjs-webpack-plugin\dist\uglify\index.js:99:22)
    at D:\pdd_web_personal\node_modules\uglifyjs-webpack-plugin\dist\uglify\index.js:105:15
    at D:\pdd_web_personal\node_modules\worker-farm\lib\farm.js:191:19
    at _combinedTickCallback (internal/process/next_tick.js:131:7)
    at process._tickCallback (internal/process/next_tick.js:180:9)
```

**If the current behavior is a bug, please provide the steps to reproduce.**
set mode production and set recordsPath and compile twice

**What is the expected behavior?**
compile success

**Other relevant information:**
webpack version:  lastest 
Node.js version:  8.10.0
Operating System: windows7
Additional tools: @ooflorent  help to see what is wrong thanks

**Comment by AMDvsTMD** [comment]
if is set config.optimization.sideEffects=false; it will be ok so does sideEffects has something wrong with AggressiveSplittingPlugin

**Comment by evilebottnawi** [comment]
@AMDvsTMD sorry for big delay, problems still exists?

**Comment by evilebottnawi** [comment]
Closing due to inactivity. Please test with latest version and feel free to reopen if still regressions. Thanks! Should be fixed in 5.4
